cmd/swapcli/main.go

```diff
@@ -451,30 +456,41 @@ func runQueryAll(ctx *cli.Context) error {
 		fmt.Printf("  Peer ID: %v\n", po.PeerID)
 		fmt.Printf("  Offers:\n")
 		for j, o := range po.Offers {
-			printOffer(o, j, "    ")
+			err = printOffer(o, j, "    ")
+			if err != nil {
+				return err
+			}
 		}
 	}
 
 	return nil
 }
 
 func runMake(ctx *cli.Context) error {
-	min := ctx.Float64(flagMinAmount)
-	if min == 0 {
-		return errNoMinAmount
+	min, err := readUnsignedDecimalFlag(ctx, flagMinAmount)
+	if err != nil {
+		return err
 	}
 
-	max := ctx.Float64(flagMaxAmount)
-	if max == 0 {
-		return errNoMaxAmount
+	max, err := readUnsignedDecimalFlag(ctx, flagMaxAmount)
+	if err != nil {
+		return err
 	}
 
-	exchangeRate := ctx.Float64(flagExchangeRate)
-	if exchangeRate == 0 {
-		return errNoExchangeRate
+	exchangeRateDec, err := readUnsignedDecimalFlag(ctx, flagExchangeRate)
+	if err != nil {
+		return err
+	}
+	exchangeRate := coins.ToExchangeRate(exchangeRateDec)
+	// TODO: How to handle this if the other asset is not ETH?
```

Comment:
this should check if the asset is an ERC20, and if it is, then read the decimals from the contract and set the `otherMin/Max` from that.

Reply:
We are in the swapcli client here.  Would we implement that as a separate API call?  I'm asking for future development, as these are not minor changes that would fit into this PR.